### ISSUE TITLE
Docker Desktop (4.6.0 on Mac M1) continues to remain in "Starting..." state, never starts up

### ISSUE THREAD

**Issue by varunshastry** [docker/for-mac#6245]
- [x ] I have tried with the latest version of Docker Desktop
  - [x ] I have tried disabling enabled experimental features
  - [ ] I have uploaded Diagnostics
  - Diagnostics ID: 20220321101221

### Expected behavior
After installing/upgrading Docker Desktop on Mac M1 chip to 4.6.0, Docker Desktop should start successfully.

### Actual behavior
After installing/upgrading Docker Desktop on Mac M1 chip to 4.6.0, Docker Desktop continues to remain in "Starting..." state.

### Information

  - macOS Version:
  - Intel chip or Apple chip: Apple chip
  - Docker Desktop Version: 4.6.0

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

% /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[FAIL] DD0031: does the Docker API work? Cannot connect to the Docker daemon at unix://Library/Containers/com.docker.docker/Data/docker.raw.sock. Is the docker daemon running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[FAIL] DD0016: is the LinuxKit VM running? vm is not running: vm has not started
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[FAIL] DD0017: can a VM be started? vm has not started: vm has not started
[PASS] DD0015: are the binary symlinks installed?
Error response from daemon: dial unix /Users/varunshastry/Library/Containers/com.docker.docker/Data/docker.raw.sock: connect: connection refused
[FAIL] DD0003: is the Docker CLI working? exit status 1
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[PASS] DD0012: is the VM networking working?
[FAIL] DD0032: do Docker networks overlap with host IPs? Cannot connect to the Docker daemon at unix://Library/Containers/com.docker.docker/Data/docker.raw.sock. Is the docker daemon running?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: can a VM be started?
    Failed with: vm has not started: vm has not started

The Docker engine runs inside a Linux VM. Therefore we must be able to start Virtual Machines.

### Steps to reproduce the behavior

  1. Navigate to Check for software updates
  2. Click "Check for software updates"  3. Install 4.6.0 
 

Attached the Diagnostics.zip file : 20220321101221.zip

Diagnostics didn't get uploaded due to the following error:
Error writing:  Put "https://docker-pinata-support.s3.amazonaws.com/incoming/3/8E346BF6-D305-4E56-9D95-9DE92494D989/20220321101258.zip": proxyconnect tcp: dial tcp :0: connect: can't assign requested address
Error uploading: Put "https://docker-pinata-support.s3.amazonaws.com/incoming/3/8E346BF6-D305-4E56-9D95-9DE92494D989/20220321101258.zip": proxyconnect tcp: dial tcp :0: connect: can't assign requested address

[20220321101221.zip](https://github.com/docker/for-mac/files/8314637/20220321101221.zip)

**Comment by daniellaera** [comment]
same issue here! please do something!

**Comment by ArsalanWahid** [comment]
Having the same issue for docker desktop  

`Docker networks overlap with host IPs? Cannot connect to the Docker daemon at unix://Library/Containers/com.docker.docker/Data/docker.raw.sock. Is the docker daemon running?`

**Comment by varunshastry** [comment]
A workaround is to uninstall v4.6.0 and install v4.5.0

**Comment by daniellaera** [comment]
hey Guys, I finally found a solution and I decided to create a video on my YT channel, hope to help others
https://www.youtube.com/watch?v=2stqWrxwBB0&t=42s

**Comment by hughsw** [comment]
Similar problem for me, running Big Sur 11.2.3. 

More detail for my case:
- I had noticed builds taking a long time to download base images...
- sometimes `docker system prune --force` would hang
- sometimes I got to the state mentioned above where the Desktop got stuck "Starting"

For me, unchecking "Use the new Virtualization framework" in the "Experimental Features" preferences fixes the problem.  (The "VirtioFS" box is greyed out because I'm not running Monterey)


Perhaps the following will help with reproducing.  The following Dockerfile typically slows down and then hangs:
```
FROM python:3.10-bullseye

RUN  true \
  && apt-get update \
  && apt-get install -y \
      apt-utils \
      emacs-nox \
      libcap-dev \
      libsndfile1-dev \
      python3 \
  && python3 -m pip install --upgrade pip \
  && true
```

Once it has hung the system, the self diagnostic took reports VM networking trouble:
```
$ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check
Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[SKIP] DD0028: is there available VM disk space?
[PASS] DD0031: does the Docker API work?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0001: is the application running?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0017: can a VM be started?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0007: is the backend responding?
[PASS] DD0014: are the backend processes running?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[FAIL] DD0012: is the VM networking working? network checks failed: failed to ping host: exit status 1
[2022-03-22T17:45:31.146061000Z][com.docker.diagnose][I] ipc.NewClient: b39b591a-diagnose-network -> Library/Containers/com.docker.docker/Data/diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[	common/pkg/diagkit/gather/diagnose/network.go:34 +0xdd
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x4d50ce0)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x43
[common/pkg/diagkit/gather/diagnose.Run.func1(0x4d50ce0)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x5a
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2, 0x4d50ce0)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x77
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x4d50ce0, 0xc0002a9828)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x36
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x100000000203000, 0xc0002a9828)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x73
[common/pkg/diagkit/gather/diagnose.walkOnce(0x46c16c0, 0xc00035f968)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0xcc
[common/pkg/diagkit/gather/diagnose.Run(0x4d50b60, 0x45fd06f, {0xc00035faf8, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x1ce
[main.checkCmd({0xc000134010, 0x6, 0x4}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:131 +0x105
[main.main()
[	common/cmd/com.docker.diagnose/main.go:97 +0x296
[2022-03-22T17:45:31.146539000Z][com.docker.diagnose][I] (85ae0c47) b39b591a-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["192.168.1.68","192.168.250.12"]}
[2022-03-22T17:45:31.732992000Z][com.docker.diagnose][E] (85ae0c47) b39b591a-diagnose-network C<-S f3ec2d96-diagnosticsd POST /check-network-connectivity (586.468518ms): failed to ping host: exit status 1
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[	common/pkg/diagkit/gather/diagnose/network.go:35 +0x15b
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x4d50ce0)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x43
[common/pkg/diagkit/gather/diagnose.Run.func1(0x4d50ce0)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x5a
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2, 0x4d50ce0)
[	common/pkg/diagkit/gather/diagnose/run.go:140 +0x77
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x4d50ce0, 0xc0002a9828)
[	common/pkg/diagkit/gather/diagnose/run.go:146 +0x36
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x100000000203000, 0xc0002a9828)
[	common/pkg/diagkit/gather/diagnose/run.go:149 +0x73
[common/pkg/diagkit/gather/diagnose.walkOnce(0x46c16c0, 0xc00035f968)
[	common/pkg/diagkit/gather/diagnose/run.go:135 +0xcc
[common/pkg/diagkit/gather/diagnose.Run(0x4d50b60, 0x45fd06f, {0xc00035faf8, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x1ce
[main.checkCmd({0xc000134010, 0x6, 0x4}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:131 +0x105
[main.main()
[	common/cmd/com.docker.diagnose/main.go:97 +0x296

[PASS] DD0032: do Docker networks overlap with host IPs?
[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?

Please investigate the following 1 issue:

1 : The test: is the VM networking working?
    Failed with: network checks failed: failed to ping host: exit status 1

VM seems to have a network connectivity issue. Please check your host firewall and anti-virus settings in case they are blocking the VM.
```

**Comment by Jorge-Eduardo-Rivadeneira** [comment]
> hey Guys, I finally found a solution and I decided to create a video on my YT channel, hope to help others
> https://www.youtube.com/watch?v=2stqWrxwBB0&t=42s

Excellent, that worked for me as well.

**Comment by mjy** [comment]
The video worked for me as well.  One thing I had to do in addition- my terminal didn't have access to the whole HD, (Security & Privacy -> Full Disk Access), this was preventing me from destroying the binary cookie file.  I had cleaned out everything but that, and no luck, missing that I hadn't delete it.   Destroying it too *finally* let a clean re-install to work.

**Comment by B-0-B-B-Y** [comment]
Tried every suggestion I could find online, couldn't get 4.6.x to work on my Mojave 12.3 M1, downgrading to 4.5.0 works perfectly, docker engine is able to start back up again.

**Comment by trungutt** [comment]
We fixed the diagnostic upload in 4.6.1 and we'd like to have more diagnostic upload to be able to look into the issue. For those who don't know how to upload diagnostic, the instruction is here https://docs.docker.com/desktop/mac/troubleshoot/#diagnose-and-feedback
